coalib/bearlib/languages/documentation/DocBaseClass.py

```diff
@@ -254,31 +256,81 @@ def extract_documentation_with_markers(content, docstyle_definition):
+
+    @staticmethod
+    def extract(content, language, docstyle):
+        """
+        Extracts all documentation texts inside the given source-code-string
+        using the coala docstyle definition files.
+
+        The documentation texts are sorted by their order appearing in
+        ``content``.
+
+        For more information about how documentation comments are
+        identified and extracted, see DocstyleDefinition.doctypes enumeration.
+
+        :param content:            The source-code-string where to extract
+                                   documentation from. Needs to be a list
+                                   or tuple where each string item is a
+                                   single line(including ending whitespaces
+                                   like ``\\n``).
+        :param language:           The programming language used.
+        :param docstyle:           The documentation style/tool used
+                                   (e.g. doxygen).
+        :raises FileNotFoundError: Raised when the docstyle definition file
+                                   was not found.
+        :raises KeyError:          Raised when the given language is not
+                                   defined in given docstyle.
+        :raises ValueError:        Raised when a docstyle definition setting
+                                   has an invalid format.
+        :return:                   An iterator returning each
+                                   DocumentationComment found in the content.
+        """
+        docstyle_definition = DocstyleDefinition.load(language, docstyle)
+        return extract_documentation_with_markers(
+                        content, docstyle_definition)
+
+    @staticmethod
+    def generate_diff(file, doc_comment, new_comment):
+        """
+        Generates diff between the original doc_comment and its fix
+        new_comment which are instances of DocumentationComment.
+
+        :param doc_comment:
+            original instance of DocumentationComment
+        :param new_comment:
+            fixed instance of DocumentationComment
+        :return:
+            diff instance
+        """
+        diff = Diff(file)
+
+        # We need to update old comment positions, as `assemble()`
+        # prepends indentation for first line.
+        old_range = TextRange.from_values(
+            doc_comment.range.start.line,
+            1,
+            doc_comment.range.end.line,
+            doc_comment.range.end.column)
+
+        # Clearing cached assemble() so a fresh one is fetched.
+        new_comment.assemble.cache_clear()
+
+        diff.replace(old_range, new_comment.assemble())
+        return diff
+
+    def process_documentation(self, *args, **kwargs):
+        """
+        Checks and handles the fixing part of documentation.
+
+        :return:
+            a tuple of processed documentation and warning_desc
```

Comment:
Just following the [PEP-257](https://www.python.org/dev/peps/pep-0257/) convention. (Even [java](http://www.oracle.com/technetwork/articles/java/index-137868.html) follows the same )
comments desc - all lower and doesn't ends with a period. :)